freestyle-config/src/main/scala/config.scala

```diff
@@ -44,6 +46,8 @@ object config {
     def load: FS[Config]
     def empty: FS[Config]
     def parseString(s: String): FS[Config]
+    def loadAs[T](implicit dec: ConfigDecoder[T]): FS[Either[DecodeError, T]]
```

Comment:
As of `0.3.1` in #377 this can also be expressed as a context bound `T: ConfigDecoder`